packages/fxa-shared/tracing/node-tracing.ts

```diff
@@ -0,0 +1,134 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { TraceExporter as GcpTraceExporter } from '@google-cloud/opentelemetry-cloud-trace-exporter';
+import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
+import { JaegerExporter } from '@opentelemetry/exporter-jaeger';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { Resource } from '@opentelemetry/resources';
+import {
+  BatchSpanProcessor,
+  ConsoleSpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+
+import { ILogger } from '../log';
+import { TracingOpts } from './config';
+
+const log_type = 'node-tracing';
+
+/**
+ * Responsible for initializing node tracing from a config object.
+ */
+export class NodeTracingInitializer {
```

Comment:
I think a comment about what is instrumented might beuseful (or in readme). I didn't know that this includes mysql and aws-sdk tracing.

Reply:
Sounds good, I'll add a link to the docs that do the auto instrumentation. This will be explicit about what all can be captured. 

The answer to the question though, this auto instrumentation includes almost everything you can think of! And maybe more in the future as that auto instrumentation package continues to expand support. 

It's possible we want to be more specific with what we instrument, but IMO, we should capture as much as possible to start out with, then reduce it as needed. There might be some traces showing up we wouldn't even aware of if we added instrumentation manually.

There are a few reasons we might not want to auto instrument though:

- In order to auto instrument, it must install a lot of packages in hopes that some of these can be instrumented. We already have an immense number of packages in FxA, so this doesn't really seem of concern.
- There might be things we simply don't care about, like dns lookups that just result in noise.
- We might want to avoid something because oit contain sensitive info that is hard to scrub.
- We might want to avoid something because it just generates too much data.